File: src/main/java/pro/sky/java/weatherapp/domain/WeatherMain.java
package pro.sky.java.weatherapp.domain;

import lombok.Data;
import lombok.NoArgsConstructor;
import pro.sky.java.weatherapp.util.WeatherUtils;

@Data
@NoArgsConstructor
public class WeatherMain {
    private Integer temp;
    private Integer pressure;
    private Integer humidity;

    public void setTemp(Float temp) {
        if (temp != null) {
            this.temp = WeatherUtils.convertToCelsius(temp);
        }
    }

    public void setPressure(Integer pressure) {
        if (pressure != null) {
            this.pressure = WeatherUtils.convertToMmHg(pressure);
        }
    }
}

File: src/main/kotlin/pro/sky/java/weatherapp/domain/WeatherMain.kt
package pro.sky.java.weatherapp.domain

import pro.sky.java.weatherapp.util.convertToCelsius
import pro.sky.java.weatherapp.util.convertToMmHg

class WeatherMain {
    var temp: Float? = null
        set(value) {
            field = value?.convertToCelsius()
        }
    var pressure: Int? = null
        set(value) {
            field = value?.convertToMmHg()
        }
    var humidity: Int? = null

    override fun toString(): String {
        return "WeatherMain(temp=$temp, pressure=$pressure, humidity=$humidity)"
    }
}